### ISSUE TITLE
docker.image.prune hangs

### ISSUE THREAD

**Issue by haruishi43** [gabrieldemarmiesse/python-on-whales#183]
Everything seems to work well, but I found that `docker.image.prune` just hangs and does nothing until `ctrl+c`.
However, I found that getting the image through `docker.image.inspect` and then removing it works.

```
# doesn't work
docker.image.prune(all=True, filter=some_label)

# work around
image = docker.image.inspect(name_of_the_image)
image.remove(prune=True)
```

Any insights on why the `prune` command isn't working correctly?

**Comment by gabrieldemarmiesse** [comment]
Hello! Could you try it without python-on-whales and report if you have the same issue? Sometimes, when you have a lot of images, the prune command takes a long time.

**Comment by haruishi43** [comment]
Thanks for the reply. I built the same image (ubuntu18:latest) with some label and ran `docker image prune -a --filter "label=some_label"`, but it works fine. It seems like it hangs when running with python-on-whales 😭

**Comment by gabrieldemarmiesse** [comment]
it is very strange as python-on-whales just calls the CLI. Can you provide more information about your system? The ouput of `docker info` would be good.

**Comment by haruishi43** [comment]
```

Server:
 Containers: 5
  Running: 5
  Paused: 0
  Stopped: 0
 Images: 189
 Server Version: 20.10.6
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 1
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runtime.v1.linux runc io.containerd.runc.v2
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: d71fcd7d8303cbf684402823e425e9dd2e99285d
 runc version: b9ee9c6314599f1b4a7f497e1f1f856fe433d3b7
 init version: de40ad0
 Security Options:
  apparmor
  seccomp
   Profile: default
 Kernel Version: 5.8.0-55-generic
 Operating System: Ubuntu 20.04.2 LTS
 OSType: linux
 Architecture: x86_64
 CPUs: 24
 Total Memory: 62.74GiB
 Name: home-pc
 ID: <REDACTED>
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
```
Some of it I redacted, but here's the list!

**Comment by gabrieldemarmiesse** [comment]
Sorry for the wait, I found the issue. The cli was asking for input and stdout wasn't redirected, resulting in a hang with no info printed on the terminal. I firxed it with this PR: https://github.com/gabrieldemarmiesse/python-on-whales/pull/204 . 

Once it's merged, I'll make a release and you'll be able to try it.

**Comment by gabrieldemarmiesse** [comment]
Can you do `pip install --upgrade python-on-whales` and tell me if it works now? The new version with the fix is 0.21.1.

**Comment by gabrieldemarmiesse** [comment]
Since I didn't get any answer, I'll assume this is working for you. Thank you for the bug report!
